pages/content/amp-dev/documentation/guides-and-tutorials/develop/custom-javascript.md

```diff
@@ -3,120 +3,141 @@ $title: Use custom JavaScript in AMP pages
+
+**1) Know what's supported.** Get to know [the rich set of supported APIs](https://github.com/ampproject/worker-dom/blob/main/web_compat_table.md). You need to think about the DOM API differently - less as a large set of properties and methods that's developed over many years, and more as a concise set of tools.
+
+**2) Use Preact.** Not only is [React](https://reactjs.org/) a popular way to build websites, but it mutates the DOM using a subset of the DOM API. `amp-script` can includes full support for this portion of the API, and thus full support for React. That said, React bundles often exceed `amp-script`'s [150K limit](../../../documentation/components/reference/amp-script.md#size-of-javascript-code). So it's recommended that you use [Preact](https://preactjs.com/), a lightweight alternative to React. Preact is designed for straightforward migration from React. With Preact, you should be able to build elaborate interactions with much less worry about what's supported. 
+
+`amp-script` has also been tested with frameworks like [Vue](https://vuejs.org/), [Angular](https://angularjs.org/), [Aurelia](https://aurelia.io/), and [lit-html](https://lit-html.polymer-project.org/), but less extensively. If you find a gap, please [file an issue](https://github.com/ampproject/worker-dom/issues) - or, better still, [submit a pull request](https://github.com/ampproject/worker-dom/pulls).
+
+Since `amp-script` cannot exhaustively support the DOM API, simply copying a library like [jQuery](https://jquery.com/) into an `<amp-script>` component will not work.
+
+## Use cases
+
+`amp-script` was created to help you fill the gaps in your AMP webpage's functionality. Since it supports a subset of the DOM and Web APIs, and since its use [comes with restrictions](https://amp.dev/documentation/components/amp-script/#restrictions), it is not an all-purpose JavaScript solution. Any substantial body of existing JavaScript is likely to need modification to work in the `amp-script` context.
+
+However, `amp-script` presents a fine way to take care of logic and interactions that existing AMP components don't provide. The following are a few excellent use cases.
 
-## DOM manipulation
+### Create new interactions
 
-For user experience and security reasons, `amp-script` enforces DOM manipulation restrictions.
+`amp-script` lets you create interactions that existing AMP components can't, opening up in AMP a door to the full creativity of the Web. That said, if an AMP component can already be leveraged to create a desired interaction, that will ultimately be simpler and more maintainable. If the AMP component simply requires some customization, you may be able to  [enhance it with `amp-script`](#enhance-amp-components). If the interaction you wish to create might be of interest to other developers, you are encouraged to [suggest and contribute](https://github.com/ampproject/amphtml/blob/master/CONTRIBUTING.md) a new component.
```

Comment:
Too much use of the word "simple" in this section and passive voice. Please rephrase.